### ISSUE TITLE
What features should it have?

### ISSUE THREAD

**Issue by elcuervo** [elcuervo/minimail#1]
This is going to be the mail thread to discuss some of the features.
It's mainly an idea harvester and not a TODO list of features.

Thanks for the feedback!
- Gmail support
- Gmail shortcuts support
- Markdown only support (rendered in html & plain md)
- Task list behaviour (http://www.mailpilot.co/)
- Zero noise.
- GPG support
- Code rendering?

From https://gist.github.com/elcuervo/9533504:

@bbonamin suggested:
- Conversation support!

@porta suggested:
- Add notes to emails.
- Gmail undo send
- Branch conversations
- Tracking (open/read/click)

@kandalf suggested:
- Cross-platform

**Comment by badosu** [comment]
Is it a desktop, web or TUI mail client?

Would it be something similar to https://en.wikipedia.org/wiki/Mailpile?

Will it fetch mails and cache them locally, like offlineimap (offlineimap is not a client, it just does this)?

Will it have a search mail feature like http://notmuchmail.org/ (ditto for offlineimap)?

**Comment by elcuervo** [comment]
@badosu Desktop. I've tried thunderbird, sparrow, mailpilot and I don't find what I want :P

**Comment by bobbyduhbrain** [comment]
I kind of hate when email conversations show reply history by default.  I think the reply chain should be hidden from the user but marked more clearly than a little arrow in the corner to open and close the reply chain.

**Comment by badosu** [comment]
I am just making so many questions because like you, I had the same urge some time ago.

Today I use mutt and offlineimap, but would love to use something simpler, minimal and useful.

Mailpile is my best hope, but I would love something better, maybe a better thunderbird? Sharing a lib with an optional TUI or a GUI?

**Comment by badosu** [comment]
It maybe could use some already developed software, like `offlineimap`, `notmuch` and have a nice interface.

I would be nice if it could be a library and you could choose whether you want to use the GUI or the TUI, like http://www.transmissionbt.com/ does.

**Comment by inkel** [comment]
I've two suggestions:
- **discourage** top posting.
- monospace fonts **only**

**Comment by badosu** [comment]
Usefult for me:
- Tree visualization of threads
- Background fetch of new emails
- Simple search
- Nice separation of folders

**Comment by inkel** [comment]
> Tree visualization of threads

**YES, PLEASE**

**Comment by sauron** [comment]
Good Idea @elcuervo.
What was always useful for me and I missed on Gmail was the ability to attach one or more email to a new one.
eg:
Email1: Discuss functionality 1
Email2: Discuss UI
Email N: Discuss functionality n 
New email: we will only implement what was discussed on Email 1 and Email N, find it attached.

Makes sense?

**Comment by badosu** [comment]
Some other ideas:
- Option to send raw text messages instead of html, ideally with a monospaced font
- Option to disable of parsing html emails (maybe using the `.mailcap` file like mutt does)
- Option to specify program to open attachments with `.mailcap`
- @sauron idea is good, but should definitely send them as attachments like a forwarded email

**Comment by btagliani** [comment]
Some suggestions:
- Fetch contact image from Social Media/Gravatar.
- Multiple accounts support.
- OSX push notifications.

**Comment by inkel** [comment]
I wonder what's the number of features that will make minimail become bloatedmail :)

**Comment by badosu** [comment]
@inkel 

Maybe if it could just receive and show messages and everything else are plugins?

**Comment by inkel** [comment]
@badosu interesting. I think I might like that.

**Comment by inkel** [comment]
In fact, @elcuervo you could make everything pluggable, following @badosu suggestion, so if people wants to have a different UI, they can use it. Or not. Your call :)

**Comment by btagliani** [comment]
@badosu @inkel 

That's a great idea!

**Comment by sauron** [comment]
"Pluggable" I would love to contribute to something like that.

**Comment by btagliani** [comment]
Something like Sublime/Atom package system?

**Comment by inkel** [comment]
I wouldn't know. I use Emacs ;)

**Comment by badosu** [comment]
@inkel @ReinventingHell @sauron 

The thing is, most users would dislike having just a raw email visualization and a one-by-one mail fetcher, which would indeed be minimal.

But this would be really ideal with you wanted something minimal, showing all the email headers, content on the visualization and only sending raw emails. Everything else should be a plugin.

**Comment by Porta** [comment]
+1 to the everything as a plugin idea.

**Comment by sauron** [comment]
Well @badosu as you mentioned, there must be a set of minimal functionalities.
The rest should be consider as a plugin or extension.
I guess the basics may be summarized with what @elcuervo proposed when opening the issue.
Perhaps, the best way to solve this is a poll asking _which of this features should be included in minimail?_, list some or all the features  and let the people vote.

**Comment by inkel** [comment]
> ...most users...

Most users won't use this, they're happy with Gmail, Mail.app, Thunderbird or whatever the heck they're using.

**Comment by badosu** [comment]
@sauron I agree with you, but I can't imagine how having gmail support can be something near to minimal.

**Comment by elcuervo** [comment]
Thank you all for the discussion. This is exactly what I wanted to happen. 
The main idea is to fill a gap that I have on email clientes.

The plugin idea is awesome and perfectly doable, specially if it's a "hackable" email client.

The inspiration behind all this is [iA Writer](http://www.iawriter.com/) and that I **always** send my emails in Markdown and I'd expect them to be read that way as well.

The initial milestones is:
- Fetch email from Gmail
- Handle conversations
- Read/Write as markdown.

I've seen some cool features listed here and I know that implementing most of them could be defined as bloated but I really want to accomplish a minimal mail experience. Like focus only in content and leave the rest to others.

I really like @inkel's suggestions about top posting and the typical font.
I have some ideas and I'll start doing some spikes on features.

The goal of this issue is to discuss ideas since after some back and forth I tend to find that some issues are not really that important or can be achieved in other way.

Again. Thanks a lot to everyone for the feedback and keep it coming. I'll add more issues with the spikes and UI ideas.

You are all awesome:

![](http://i.imgur.com/s0HhCmB.gif)

**Comment by rogerwang** [comment]
search function like `bup` or `notmuch`: notmuchmail.org

**Comment by Porta** [comment]
@elcuervo Have you chosen a language?

**Comment by elcuervo** [comment]
@Porta I have several ideas but I'm pretty inclined to node-webkit since it's easily customizable and works pretty well in several platforms.

**Comment by Porta** [comment]
@elcuervo Cool. This might be the excuse I needed to learn Node.
